int32 Mask
int32 DesiredDepth
---
int32 CurrentDesiredDepth
